pkg/test/server.go

```diff
@@ -0,0 +1,156 @@
+package test
+
+import (
+	"encoding/csv"
+	"fmt"
+	"log"
+	"net"
+	"os"
+
+	"github.com/miekg/dns"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/sync/errgroup"
+)
+
+var domainToIP = loadRecords("data/test-300.csv")
+
+const truncateSize = 512
+
+type Server struct {
+	Addr    string
+	TCPPort int
+	UDPPort int
+}
+
+func (s *Server) parseUDPAddr() string {
```

Comment:
There's no parsing involved here; maybe just call this `udpAddr()`? (Same with the TCP version.)
Or just call `net.JostHostPort()` directly and skip the wrapper method?